### ISSUE TITLE
Try correcting the name to the name of an existing getter, or defining a getter or field named 'surfaceContainerHighest'

### ISSUE THREAD

**Issue by suhailthakrani** [syncfusion/flutter-widgets#1875]
### Bug description

/C:/Users/HC/AppData/Local/Pub/Cache/hosted/pub.dev/syncfusion_flutter_datepicker-25.2.5/lib/src/date_picker/theme.dart:83:55: Error: The getter 'surfaceContainerHighest' isn't defined for the class 'ColorScheme'.
 - 'ColorScheme' is from 'package:flutter/src/material/color_scheme.dart' ('/C:/src/flutter/packages/flutter/lib/src/material/color_scheme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'surfaceContainerHighest'.
  Color? get weekNumberBackgroundColor => colorScheme.surfaceContainerHighest;
                                                      ^^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDebug'.
> Process 'command 'C:\src\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 19s
Error: Gradle task assembleDebug failed with exit code 1

### Steps to reproduce

Just used the SfDateRangePicker and got this error

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:syncfusion_flutter_datepicker/datepicker.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: Text('Syncfusion Datepicker')),
        body: Center(
          child: SfDateRangePicker(),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![image](https://github.com/syncfusion/flutter-widgets/assets/57296007/f2c46af3-358e-46d9-8b77-cd471affb056)

</details>

### Stack Traces

<details open><summary>Stack Traces</summary>

```dart
/C:/Users/HC/AppData/Local/Pub/Cache/hosted/pub.dev/syncfusion_flutter_datepicker-25.2.5/lib/src/date_picker/theme.dart:83:55: Error: The getter 'surfaceContainerHighest' isn't defined for the class 'ColorScheme'.
 - 'ColorScheme' is from 'package:flutter/src/material/color_scheme.dart' ('/C:/src/flutter/packages/flutter/lib/src/material/color_scheme.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'surfaceContainerHighest'.
  Color? get weekNumberBackgroundColor => colorScheme.surfaceContainerHighest;
                                                      ^^^^^^^^^^^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception

```

</details>

### On which target platforms have you observed this bug?

Android

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.19.3, on Microsoft Windows [Version 10.0.19045.4412], locale en-US)
    • Flutter version 3.19.3 on channel stable at C:\src\flutter             
    • Upstream repository https://github.com/flutter/flutter.git             
    • Framework revision ba39319843 (3 months ago), 2024-03-07 15:22:21 -0600
    • Engine revision 2e4ba9c6fb                                             
    • Dart version 3.3.1                                                     
    • DevTools version 2.31.1                                                
                                                                             
[√] Windows Version (Installed version of Windows is version 10 or higher)   

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at C:\Users\HC\AppData\Local\Android\Sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = C:\Users\HC\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 2022.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-b2043.56-10027231)

[√] VS Code (version 1.89.1)
    • VS Code at C:\Users\HC\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.89.20240501

[√] Connected device (3 available)
    • Infinix X688B (mobile) • 061972516G011808 • android-arm64  • Android 11 (API 30)
    • Chrome (web)           • chrome           • web-javascript • Google Chrome 124.0.6367.208
    • Edge (web)             • edge             • web-javascript • Microsoft Edge 125.0.2535.51

[√] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by loud1y** [comment]
got same err

**Comment by LokeshPalani** [comment]
Hi @loud1y | @suhailthakrani,

We have updated our packages to be compatible with the latest Flutter SDK v3.22.0. In this version, Flutter introduced some breaking changes that were addressed in our packages. This may cause issues on your end if you are using a lower version of the Flutter SDK. So, we kindly request you to upgrade your Flutter SDK version to 3.22.0 or else use the syncfusion_flutter_datepicker package version below [25.2.4](https://pub.dev/packages/syncfusion_flutter_datepicker/versions/25.2.4).

We hope it will help you to resolve the issue. If you are still facing issue, we recommend that you check by clearing the cache using the below mentioned methods:

- ​In the Flutter terminal, add the "flutter pub cache clean" command to clear all the cache.
- ​Close the running application and then add the "flutter clean" command.
- ​Delete the “pubspec.lock” file.
- ​Afterwards, add the "flutter pub get" command to get all the dependencies in the pubspec.yaml file.

 

Regards,

Lokesh P.

**Comment by LokeshPalani** [comment]
Hi @loud1y | @suhailthakrani,

We have updated our packages to be compatible with the latest Flutter SDK v3.22.0. In this version, Flutter introduced some breaking changes that were addressed in our packages. This may cause issues on your end if you are using a lower version of the Flutter SDK. So, we kindly request you to upgrade your Flutter SDK version to 3.22.0 or else use the syncfusion_flutter_datepicker package version below [25.2.4](https://pub.dev/packages/syncfusion_flutter_datepicker/versions/25.2.4).

We hope it will help you to resolve the issue. If you are still facing issue, we recommend that you check by clearing the cache using the below mentioned methods:

- ​In the Flutter terminal, add the "flutter pub cache clean" command to clear all the cache.
- ​Close the running application and then add the "flutter clean" command.
- ​Delete the “pubspec.lock” file.
- ​Afterwards, add the "flutter pub get" command to get all the dependencies in the pubspec.yaml file.

 
Regards,
Lokesh P.

**Comment by jlsimmo** [comment]
Hi @LokeshPalani  - I'm new to flutter (I use FlutterFlow) and I have exactly the same problem as described above... 
since i use Flutterflow, I'm not familiar with the 'flutter terminal' you mention above - where can I find it please?\Thanks,\John

**Comment by rob-mur** [comment]
Can confirm downgrading to 25.2.4 worked on my end. 

I think it would have been nicer if this breaking change came through in a major version change for this package to avoid breaking builds, but the fix is trivial so no worries.

**Comment by jlsimmo** [comment]
@rob-mur  - Rob, lucky you... I'm using Flutterflow and can't find out how to downgrade to 25.2.4 
I'm still stuck - over 48 hours without being able to do anything on my project because of this. I agree with your comments about major changes

**Comment by LokeshPalani** [comment]
Hi @jlsimmo,

We would like to inform that the Flutter Flow using Flutter SDK v3.19.1. Our packages are now compatible with v3.22.0. To resolve the issue, we suggest you use the syncfusion_flutter_datepicker package version below [25.2.4](https://pub.dev/packages/syncfusion_flutter_datepicker/versions/25.2.4) without caret (^) sign to prevent upgrading the Flutter SDK to the latest version in the Flutter Flow.

Regards,
Lokesh P.

**Comment by suhailthakrani** [comment]
> Hi @loud1y | @suhailthakrani,
> 
> We have updated our packages to be compatible with the latest Flutter SDK v3.22.0. In this version, Flutter introduced some breaking changes that were addressed in our packages. This may cause issues on your end if you are using a lower version of the Flutter SDK. So, we kindly request you to upgrade your Flutter SDK version to 3.22.0 or else use the syncfusion_flutter_datepicker package version below [25.2.4](https://pub.dev/packages/syncfusion_flutter_datepicker/versions/25.2.4).
> 
> We hope it will help you to resolve the issue. If you are still facing issue, we recommend that you check by clearing the cache using the below mentioned methods:
> 
> * ​In the Flutter terminal, add the "flutter pub cache clean" command to clear all the cache.
> * ​Close the running application and then add the "flutter clean" command.
> * ​Delete the “pubspec.lock” file.
> * ​Afterwards, add the "flutter pub get" command to get all the dependencies in the pubspec.yaml file.
> 
> Regards, Lokesh P.

The issue has been resolved on the latest version of Flutter 3.22. Thank you so much @LokeshPalani

**Comment by suhailthakrani** [comment]
Thank you so much. The issue has been resolved by updating the flutter to the latest version.
